servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpSetCookie.java

```diff
@@ -462,27 +480,34 @@ private enum ParseState {
         Unknown
     }
 
-    private static final class ParseStateCharSequence implements CharSequence {
-        final ParseState state;
-
-        ParseStateCharSequence(final ParseState state) {
-            this.state = state;
-        }
-
-        @Override
-        public int length() {
-            throw new UnsupportedOperationException();
-        }
-
-        @Override
-        public char charAt(final int index) {
-            throw new UnsupportedOperationException();
-        }
-
-        @Override
-        public CharSequence subSequence(final int start, final int end) {
-            throw new UnsupportedOperationException();
+    private static ParseState parseStateOf(CharSequence fieldName) {
+        // Try a binary search based on length. We can read length without bounds checks.
+        int len = fieldName.length();
+        if (len >= 4 && len <= 8) {
+            if (len < 7) {
+                if (len == 4) {
+                    if (contentEqualsIgnoreCase("path", fieldName)) {
+                        return ParseState.ParsingPath;
+                    }
+                } else if (len == 6 && contentEqualsIgnoreCase("domain", fieldName)) {
```

Comment:
Is checking `len` give visible advantage? Impl of `contentEqualsIgnoreCase` checks `a.length() != b.length()` before starting comparison of each character

Reply:
An earlier version used just `contentEqualsIgnoreCase`, and using the length for the branch tree like this was definitely faster. This `parseStateOf` method is some of the hottest code in the parser.